tools/sdk-versioner/src/main.rs

```diff
@@ -263,23 +341,26 @@ mod tests {
 
     #[test]
     fn update_dependencies_with_versions_and_paths() {
-        test_with_opt(
-            Args {
-                crate_paths: Vec::new(),
-                sdk_path: Some("/foo/asdf/".into()),
-                sdk_version: Some("0.5.0".to_string()),
-                smithy_version: Some("0.35.0".to_string()),
+        test_with_context(
+            DependencyContext {
+                sdk_path: Some(&PathBuf::from("/foo/asdf/")),
+                versions_manifest: Some(versions_toml_for(&[
+                    ("aws-config", "0.5.0"),
+                    ("aws-sdk-s3", "0.13.0"),
+                    ("aws-smithy-types", "0.10.0"),
+                    ("aws-smithy-http", "0.9.0"),
+                ])),
             },
-        br#"
+            br#"
             [package]
             name = "test"
             version = "0.1.0"
 
             [dependencies]
             aws-config = { version = "0.5.0", path = "/foo/asdf/aws-config" }
-            aws-sdk-s3 = { version = "0.5.0", path = "/foo/asdf/s3" }
-            aws-smithy-types = { version = "0.35.0", path = "/foo/asdf/aws-smithy-types" }
-            aws-smithy-http = { version = "0.35.0", path = "/foo/asdf/aws-smithy-http", features = ["test-util"] }
+            aws-sdk-s3 = { version = "0.13.0", path = "/foo/asdf/s3" }
+            aws-smithy-types = { version = "0.10.0", path = "/foo/asdf/aws-smithy-types" }
+            aws-smithy-http = { version = "0.9.0", path = "/foo/asdf/aws-smithy-http", features = ["test-util"] }
```

Comment:
It may be useful to add some major and patch versions in here just to increase the cases it's covering

Reply:
The version numbers themselves are opaque to the tool, so I'm not too worried about this. Just needs to test that they're coming from the `versions.toml` file.